packages/app/src/main.tsx

```diff
@@ -3,9 +3,12 @@ import { createRoot } from 'react-dom/client';
 import { BrowserRouter } from 'react-router-dom';
 
 import { App } from './App';
+import { Sidebar } from './systems/Core/components/Sidebar';
 
 createRoot(document.getElementById('root')!).render(
   <BrowserRouter basename={import.meta.env.BASE_URL}>
-    <App />
+    <Sidebar>
```

Comment:
I'm kinda confused here, why do we need the `Sidebar` wrapping the entire application? 🤔

Reply:
You can just add it inside the `Topbar` component, this will avoid re-renders at the top of the app and all render triggers will be applied just for the `Topbar` itself. You really don't need this as a wrapper.

Reply:
If you want to get the ref from the `Layout` you can pass the ref created inside the `Layout` in the layout context, already as some props there, you can use it to avoid this wrapper stuff.